src/actions.js

```diff
@@ -18,6 +18,38 @@ export const connectionError = () => {
 
 const fetchCache = {};
 
+const getJsonOrText = (path, response, useCache, method = 'GET') => response.text().then(
+  (textData) => {
+    let data = textData;
+    if (data) {
+      try {
+        data = JSON.parse(data);
+        if (useCache && method === 'GET' && response.status === 200) {
+          fetchCache[path] = textData;
+        }
+      } catch (e) {
+        // # do nothing
+      }
+    }
+    return { response, data, status: response.status, headers: response.headers };
+  });
+
+export const fetchAuth0 = (opts) => {
+  const { path, method = 'GET', dispatch } = opts;
```

Comment:
Hey Thanh,

It would be cool to patch `fetchAuth0` so that it does not accept a `path` or `method` - it just always does the `auth` request to fence.  

After making that change - you can wire up `fetchAuth0` with a mutex, so that only one is running at a time - ex:
```
var pendingRequest = null;
function fetchAuth0() {
   if ( pendingRequest ) { return pendingRequest; }
  pendingRequest = fetch( ... ).then( 
     function() { pendingRequest = null; ... },
     function() { pendingRequest = null; ... }
  );
  return pendingRequest;
}
```
That way multiple simultaneous overlapping async fetch requests will only result in a single Auth0 request that they'll all wait for.  What do you think?